packages/botonic-plugin-contentful/src/contentful/delivery-api.ts

```diff
@@ -84,3 +99,93 @@ export class AdaptorDeliveryApi implements DeliveryApi {
     return query
   }
 }
+
+export interface ContentWithNameFields {
+  // The content code (eg. PRE_FAQ1) Not called Id to differentiate from contentful automatic Id
+  name: string
+}
+
+export interface CommonEntryFields extends ContentWithNameFields {
+  // Useful to display in buttons or reports
+  shortText?: string
+  keywords?: string[]
+  searchableBy?: contentful.Entry<SearchableByKeywordsFields>[]
+  partitions?: string[]
+  dateRange?: contentful.Entry<DateRangeFields>
+  followup?: contentful.Entry<FollowUpFields>
+}
+export type FollowUpFields = CommonEntryFields
+
+export class ContentfulEntryUtils {
+  static getContentId<T extends ContentType = ContentType>(
+    entry: contentful.Entry<any>
+  ): ContentId {
+    return ContentId.create(
+      ContentfulEntryUtils.getContentModel(entry),
+      entry.sys.id
+    )
+  }
+
+  /**
+   * Will be false for broken references, or when we have only fetched
+   * the full Entry tree
+   */
+  static isFullEntry(entry: contentful.Entry<any>): boolean {
+    return !!entry.fields
+  }
+
+  static getContentModel<T extends ContentType = ContentType>(
+    entry: contentful.Entry<any>
+  ): T {
+    // https://blog.oio.de/2014/02/28/typescript-accessing-enum-values-via-a-string/
+    if (!entry.sys.contentType) {
+      throw new CmsException(
+        `Entry '${entry.sys.id}' not fully loaded or referencing a deleted content`
+      )
+    }
+    const typ = entry.sys.contentType.sys.id
```

Comment:
Better to put `return entry.sys.contentType.sys.id as T` ?

Reply:
done in 917cf23